src/lib.rs

```diff
@@ -126,15 +126,55 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateName = String;
+pub trait TemplateRepository {
+    fn read_template(&self, path: &TemplateName) -> Result<String>;
```

Comment:
Drat, clippy isn't happy about TemplateName references.

Could we change TemplateName to be a typdef for str?  If not, I'm fine getting rid of the typedef for now.